File: Swift/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.swift
import Foundation

func solution(_ babbling:[String]) -> Int {
    let canSpeakWords = ["aya", "ye", "woo", "ma"]
    var count = 0
    
    for word in babbling {
        var tempWord = word
        var possible = true
        
        for speakWord in canSpeakWords {
            if let range = tempWord.range(of: speakWord) {
                tempWord.replaceSubrange(range, with: " ")
            }
        }
        
        if tempWord.trimmingCharacters(in: .whitespaces).isEmpty {
            count += 1
        }
    }
    
    return count
}

File: Java/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        
        for (String word : babbling) {
            String replaced = word.replaceAll("aya|ye|woo|ma", " ");
            
            if (replaced.replaceAll(" ", "").equals("")) {
                answer ++;
            }
        }
        return answer;
    }
}